catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandlerTest.java

```diff
@@ -0,0 +1,148 @@
+  private QueryResponseTransformer mockServiceReference;
+  private ServletOutputStream mockServletOutputStream;
+  private HttpServletResponse mockHttpServiceResponse;
+
+  private static final String GZIP = "gzip";
+  private static final String QUERY_PARAM = ":transformerId";
+  private static final String RETURN_ID = "kml";
+  private static final String MIME_TYPE = "application/vnd.google-earth.kml+xml";
+  private static final String SAFE_BODY =
+      "{\"src\":\"ddf.distribution\",\"start\":1,\"count\":250,\"cql\":\"anyText ILIKE '*'\",\"sorts\":[{\"attribute\":\"modified\",\"direction\":\"descending\"}],\"id\":\"7a491439-948e-431b-815e-a04f32fecec9\"}";
+  private static final String CONTENT = "test";
+  private static final String SERVICE_NOT_FOUND = "{\"message\":\"Service not found\"}";
+  private static final String SERVICE_SUCCESS = "";
+
+  @Before
+  public void setUp() throws Exception {
+    initMocks(this);
+
+    mockBundleContext = mock(BundleContext.class);
+    queryResponseTransformers = new ArrayList<>();
+    mockQueryResponseTransformer = mock(ServiceReference.class);
+    when(mockQueryResponseTransformer.getProperty(Core.ID)).thenReturn(RETURN_ID);
+
+    MimeType mimeType = new MimeType(MIME_TYPE);
+    mockBinaryContent =
+        new BinaryContentImpl(new ByteArrayInputStream(CONTENT.getBytes()), mimeType);
+
+    queryResponseTransformers.add(mockQueryResponseTransformer);
+    cqlTransformHandler = new CqlTransformHandler(queryResponseTransformers, mockBundleContext);
+
+    mockEndpointUtil = mock(EndpointUtil.class);
+    mockRequest = mock(Request.class);
+    mockResponse = mock(Response.class);
+
+    when(mockEndpointUtil.safeGetBody(mockRequest)).thenReturn(SAFE_BODY);
+
+    mockCqlQueryResponse = mock(CqlQueryResponse.class);
+
+    when(mockEndpointUtil.executeCqlQuery(any(CqlRequest.class))).thenReturn(mockCqlQueryResponse);
+
+    cqlTransformHandler.setEndpointUtil(mockEndpointUtil);
+
+    mockQueryResponse = mock(QueryResponse.class);
+    when(mockCqlQueryResponse.getQueryResponse()).thenReturn(mockQueryResponse);
+
+    mockServiceReference = mock(QueryResponseTransformer.class);
+    when(mockBundleContext.getService(mockQueryResponseTransformer))
+        .thenReturn(mockServiceReference);
+    when(mockServiceReference.transform(mockQueryResponse, Collections.emptyMap()))
+        .thenReturn(mockBinaryContent);
+
+    mockServletOutputStream = mock(ServletOutputStream.class);
+
+    mockHttpServiceResponse = mock(HttpServletResponse.class);
+
+    when(mockResponse.raw()).thenReturn(mockHttpServiceResponse);
+
+    when(mockHttpServiceResponse.getOutputStream()).thenReturn(mockServletOutputStream);
+  }
+
+  @Test
+  public void testNoServiceFound() throws Exception {
+    when(mockRequest.params(QUERY_PARAM)).thenReturn("xml");
+
+    String res = (String) cqlTransformHandler.handle(mockRequest, mockResponse);
+
+    assertThat(res, is(SERVICE_NOT_FOUND));
```

Comment:
❓ Can we also test the status code here too? Should be `404 Not Found` right? Also should we write different tests to track all of the possible status codes that this endpoint can produce?